src/libraries/Microsoft.PowerFx.Core/Texl/Builtins/Lookup.cs

```diff
@@ -47,6 +48,20 @@ public override bool CheckInvocation(TexlBinding binding, TexlNode[] args, DType
             // The return type is dictated by the last argument (projection) if one exists. Otherwise it's based on first argument (source).
             returnType = args.Length == 2 ? argTypes[0].ToRecord() : argTypes[2];
 
+            // Ensure that the arg at index 1 is boolean or can be coersed.
+            if (CheckType(args[1], argTypes[1], DType.Boolean, DefaultErrorContainer, out var matchedWithCoercion) || DType.Boolean.Accepts(argTypes[1]))
```

Comment:
Doesn't CheckType cover the Dtype.Boolean.Accepts case?

Reply:
You are correct, I removed it.